### PR TITLE
Matomo: Track interpage navigation

### LINKED ISSUE #495: Analytics: Use Next.js <Script> component
Low priority: We currently use a native <script> and dangerouslySetInnerHTML for Matomo and Google Analytics. Let's see if we can use Next's provided component to get these scripts in the head.

### LINKED ISSUE #595: Matomo: Track navigation between pages
## Problem

We track when a user first lands on a collection page, but we don't track their navigation between pages because Matomo doesn't know about Next.js routing.

## Do this

Use a strategy similar to 60654522b85213e23d49f9b7354251623c2c91a2 to track when the Next route changes in Matomo.

## Background

https://github.com/iFixit/react-commerce/pull/594/commits/60654522b85213e23d49f9b7354251623c2c91a2#r939062850

CC @davidrans

### PR DESCRIPTION
**Pull request by masonmcelvain**
- Tracks navigation between pages in Matomo.
- Uses Next's `Script` component to run the Matomo script after the page is interactive. Consequently, the script is moved from the head to the body.

## QA
- Make sure the preview link you're using is added to matomo.ubreakit.com (see screenshot below)
- Click around through some links in the preview.
- Verify that the visits are recorded in the `Visits Log` in matomo's dashboard.
- Verify that navigation on `main` wasn't working, but it's working on this branch.
- Thanks! 🙏🏻 

image

Closes #595 
Closes #495

### CODE REVIEW

**Comment by masonmcelvain** [comment]
dev_block ✌🏻

**Comment by masonmcelvain** [comment]
un_dev_block ✌🏻 Seems to be working

image

**Comment by jordycosta** [comment]
QA 🟢 

Navigation is now being tracked past /Parts and /Tools 👍 

---

### Testing Procedure:
1. Navigate to the [Vercel preview](https://react-commerce-prod-git-matomo-track-interpage-nav-ifixit.vercel.app/Parts)
2. Click on some model collections pages
3. Ensure that these are being tracked on [matomo](https://matomo.ubreakit.com/)

---

<details>
<summary> View Visits Log </summary> <br>

action_log_branch
</details>

File: frontend/components/analytics/Matomo.tsx [hunk 1]
```diff
@@ -1,27 +1,43 @@
...
+declare const _paq: string[][] | undefined;
 
 export function Matomo() {
+   const router = useRouter();
+   React.useEffect(() => {
+      const handleRouteChange = (url: string) => {
+         if (typeof window !== 'undefined' && _paq) {
+            _paq.push(['setCustomUrl', url]);
+            _paq.push(['setDocumentTitle', document.title]);
+            _paq.push(['trackPageView']);
```
Comment (jarstelfox) [inline frontend/components/analytics/Matomo.tsx:15]: Hey I think we should look at doing something similar to: https://github.com/iFixit/ifixit/commit/7fa161dee14bd23df385f4129994f39627955ee3

Just adding some types and nice function calls around Matomo so we never need to worry about `_paq` and how it works

Comment (masonmcelvain) [reply frontend/components/analytics/Matomo.tsx:15]: Good idea!

**Comment by masonmcelvain** [comment]
dev_block ✌🏻 on wrapping `_paq`

**Comment by masonmcelvain** [comment]
un_dev_block ✌🏻

**Comment by jordycosta** [comment]
QA 🟢 

Followed testing procedure shown on https://github.com/iFixit/react-commerce/pull/598#issuecomment-1208530904

tracking

**Comment by sterlinghirsh** [comment]
CR :zap:

**Review by sterlinghirsh — APPROVED** [review]

**Pull request merged**